Extended documentation for upgrading from CI3 to CI4 (#4338)
codeigniter4/CodeIgniter4: user_guide_src/source/installation/upgrade_configuration.rst

Upgrade Configuration

> ###############################################################################

Reviewer: Please don't over-extend the lines here for this file.

.. contents::
    :local:
    :depth: 1

Documentations
============================================================

- `Config Documentation Codeigniter 3.X <http://codeigniter.com/userguide3/libraries/config.html>`_

> - `Configuration Documentation Codeigniter 4.X <http://codeigniter.com/user_guide/general/configuration.html>`_

Reviewer 2: ```suggestion
- `Configuration Documentation Codeigniter 4.X </general/configuration.html>`_
```

What has been changed
============================================================

> - In CI4 the configuration is stored in classes which extend ``CodeIgniter\Config\BaseConfig``.
> - Within the configuration class the config values are stored in public class variables.
> - The method to fetch config values has been changed

Reviewer: ```suggestion
- In CI4, the configurations are now stored in classes which extend ``CodeIgniter\Config\BaseConfig``.
- Within the configuration class, the config values are stored in public class properties.
- The method to fetch config values has been changed.
```

Upgrade Guide
============================================================
1. You have to change the values in the default CI4 config files according to the changes in the CI3 files. The config names are pretty much the same as in CI3.
2. If you are using custom config files in your CI3 project you have to create those files as new php classes in your CI4 project in ``app/Config/``. These classes should be in the ``Config`` namespace and should extend ``CodeIgniter\Config\BaseConfig``.

> 3. Once you have created all custom config classes you have to copy the variables from the CI3 config into the new CI4 config class as public class variable.

Reviewer: ```suggestion
3. Once you have created all custom config classes, you have to copy the variables from the CI3 config into the new CI4 config class as public class properties.
```

> 4. Now you have to change the config fetching syntax everywhere you fetch config values. The CI3 syntax is something like ``$this->config->item('item_name');``. You have to change this into ``config('MyConfigFile')->item_name``.

Reviewer: ```suggestion
4. Now, you have to change the config fetching syntax everywhere you fetch config values. The CI3 syntax is something like ``$this->config->item('item_name');``. You have to change this into ``config('MyConfigFile')->item_name;``. Alternatively, you can use the object-oriented approach: ``CodeIgniter\Config\Config::get('MyConfigFile')->item_name;``
```
